multicast/src/main/kotlin/com/dropbox/flow/multicast/ChannelManager.kt

```diff
@@ -51,172 +55,183 @@ internal class ChannelManager<T>(
     private val onEach: suspend (T) -> Unit,
 
     private val upstream: Flow<T>
-) : StoreRealActor<ChannelManager.Message<T>>(scope) {
-    private val buffer = Buffer<T>(bufferSize)
-    /**
-     * The current producer
-     */
-    private var producer: SharedFlowProducer<T>? = null
-
-    /**
-     * Tracks whether we've ever dispatched value or error from the current producer.
-     * Reset when producer finishes.
-     */
-    private var dispatchedValue: Boolean = false
-
-    /**
-     * List of downstream collectors.
-     */
-    private val channels = mutableListOf<ChannelEntry<T>>()
-
-    override suspend fun handle(msg: Message<T>) {
-        when (msg) {
-            is Message.AddChannel -> doAdd(msg)
-            is Message.RemoveChannel -> doRemove(msg.channel)
-            is Message.Dispatch.Value -> doDispatchValue(msg)
-            is Message.Dispatch.Error -> doDispatchError(msg)
-            is Message.Dispatch.UpstreamFinished -> doHandleUpstreamClose(msg.producer)
-        }
-    }
+) {
 
     suspend fun addDownstream(channel: SendChannel<Message.Dispatch.Value<T>>) =
-        send(Message.AddChannel(channel))
+        actor.send(Message.AddChannel(channel))
 
     suspend fun removeDownstream(channel: SendChannel<Message.Dispatch.Value<T>>) =
-        send(Message.RemoveChannel(channel))
+        actor.send(Message.RemoveChannel(channel))
 
-    /**
-     * Called when the channel manager is active (e.g. it has downstream collectors and needs a
-     * producer)
-     */
-    private fun newProducer() = SharedFlowProducer(scope, upstream, ::send)
+    suspend fun close() = actor.close()
+
+    private val actor = Actor()
```

Comment:
Nitpick fields before functions

Reply:
i tend to go the other way in this case - public interface first. then actor field with inner class for better readability